### PR TITLE
REV-2099: Add an expanded PR template

### PR DESCRIPTION
**Pull request by inventhouse**
Add an expanded PR template that also directs edX engineers to instructions for deploying their changes

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+changes.
+
+## Supporting information
+
+Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
+Be sure to check they are publicly readable, or if not, repeat the information here.
+
+## Testing instructions
+
+Please provide detailed step-by-step instructions for testing this change.
```
Comment (jhan217) [inline .github/PULL_REQUEST_TEMPLATE.md:30]: Do we want to ask "how did YOU test this change?"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Other information
+
+Include anything else that will help reviewers and consumers understand the change.
+- Does this change depend on other changes elsewhere?
+- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
```
Comment (jhan217) [inline .github/PULL_REQUEST_TEMPLATE.md]: OpenEdx vs edx.org differences?
Development vs production environment differences?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Please give the pull request a short but descriptive title.
+
+Use this template as a guide. Omit sections that don't apply. You may link to information rather than copy it.
+
+edX Internal Developers: Please also follow these steps to integrate your changes https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories
+-->
+
+## Description
+
+Describe what this pull request changes, and why. Include implications for people using this change.
```
Comment (jhan217) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: "and why these changes were made"? 

🎱 Personally prefer a more active tone so that the audience feels compelled to take action. So more "what did you do and why did you do it" style wording.

Comment (jhan217) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: 🎱 `Include implications for people using this change.` -> `How will these changes affect other [people|installations of edx|etc.]?`

**Review by dianekaplan — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Design decisions and their rationales should be documented in the repo (docstring / ADR), per
```
Comment (jhan217) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: 🎱 `Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](link).`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<!--
+Please give the pull request a short but descriptive title.
+
+Use this template as a guide. Omit sections that don't apply. You may link to information rather than copy it.
+
+**edX Internal Developers:** Please also follow these steps to integrate your changes https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories
```
Comment (jhan217) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Looks like this is a duplicate of https://github.com/edx/frontend-app-payment/pull/476/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35R38? Should we get rid of one?

Comment (inventhouse) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: The duplication is deliberate; the ticket is specifically "address issue of commits being merged and not deployed to production" and, while I want to make the PR template apply to internal and external developers, I don't want that message to get lost (@dianekaplan's feedback was "Can we beat them over the head with it?" 😄 )

Comment (jhan217) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: Sure, just keep in mind that the longer + more requests you have the more likely people's eyes will glaze over/miss stuff/skip it entirely

Comment (inventhouse) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: I see your point about potential confusion
